enhancements/console/customize-perspectives.md

```diff
@@ -0,0 +1,272 @@
+---
+title: customize-perspectives
+authors:
+  - "@lokanandaprabhu"
+reviewers:
+  - "@jerolimov"
+  - "@spadgett"
+approvers:
+  - "@spadgett"
+  - "@cvogt"
+api-approvers:
+  - "@jerolimov"
+  - "@cvogt"
+creation-date: 2022-07-26
+last-updated: 2022-08-02
+tracking-link: 
+  - https://issues.redhat.com/browse/ODC-6695
+status: implementable
+---
+
+# Customize Perspectives 
+
+## Summary
+
+This document provides details about how cluster administrator 
+can hide user perspectives. 
+
+## Motivation
+
+Cluster administrator wants to hide the "Administrator" perspective 
+from non-privileged or all users and to hide the "Developer" 
+perspective for all users.
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin, I want to hide "Administrator" perspective 
+from all non-privileged users. Non-privileged users are users that cannot get
+all `namespaces` (based on RBAC check)
+
+#### Story 2
+
+As a cluster admin, I want to hide "Administrator" perspective
+for all users.
+
+#### Story 3
+
+As a cluster admin, I want to hide "Developer" perspective
+for all users.
+
+### Goals
+
+- Allow cluster admins to hide the user perspectives.
+- Provide a way that cluster admins can do this based on resource permissions (RBAC).
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console` CRD. 
+We can add 
+`spec.customization.perspectives` which can have list of customization details.
+
+### Workflow Description
+
+To disable perspective for users with limited permissions we need a list of options with a perspective name and the required permissions.
+
+```text
+├── Spec
+│   ├── customization
+|       ├── perspectives 
+|           ├── name
+|           ├── requiresAccessReview
+|               ├── kind
+|               ├── verb
+└── ...
+```
+
+This might be extendable in the future if we want replace (hide one, show another) a perspective based on a resource permissions:
+
+```text
+├── Spec
+│   ├── customization
+|       ├── perspectives 
+|           ├── name
+|           ├── requiresAccessReview
+|           |   ├── kind
+|           |   ├── verb
+|           ├── missingAccessReview
+|               ├── kind
+|               ├── verb
+└── ...
+```
+
+Each perspective conforms to the following schema:
+
+- `disabled?: boolean` , define if the perspective is never shown
```

Comment:
Booleans are discouraged in APIs, so we'd want an enum value instead.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types

Reply:
I would use "(optional)" here instead of "?" as not everyone is familiar with TypeScript syntax.

Reply:
Hey @spadgett, good hint with the boolean. I see at least three options:

1. Keep boolean as it is still used for other enabled attributes, etc.
2. Just use `disabled` with `True` and `False` strings (defined as enum) for now. Similar to `Status corev1.ConditionStatus` but without `Unknown`
3. Or also switch the key to something that give us more options in the future. Enabled? Status? Visible?

For me personally everything is fine. Sam, should we go with number two for now? A disabled enum with `True` and `False`?

Reply:
I suspect it won't pass API review if we leave it a boolean, even if we have other boolean flags in the config. I'd suggest something like `state: {Enabled,Disabled}` for now, and we can work with the API reviewers on the best option when we have the openshift/api PR open.